com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/plugin/PluginScanner.java

```diff
@@ -60,6 +76,9 @@
 				if (pluginBaseClass.isAssignableFrom(klass) && !isAbstract && !isPrivate) {
 					Bob.verbose("Found plugin " + className);
 					plugins.add((T)klass.newInstance());
+					if (plugins.size() > 1) {
+						throw new CompileExceptionError("PluginScanner found more than one class implementing " + pluginBaseClass);
```

Comment:
In general, I think it's better form to check if we should return first, before creating more resources.
I think it applies here as well, why not check if the list already has one item and then throw?